### DISCUSSION TITLE
Using Singularity containers on Supercomputer

### DISCUSSION THREAD

**Discussion by amir-rakhimov** [nig-sc/nigsc_homepage2#300] [Q&A] [answered]
I am using the NIG supercomputer for RNA-seq data analysis. I want to use tools like samtools using Singularity containers. However, I am not sure which version to use.

When I search for samtools-1.9, there are several containers (I am showing a few):

> samtools:1.9--h10a08f8_12
> samtools:1.9--h43f6869_9
> samtools:1.9--h46bd0b3_0
> samtools:1.9--h57cc563_6
> samtools:1.9--h57cc563_7
> samtools:1.9--h8ee4bcc_1

These containers are all v1.9, yet they have specific IDs. What is the difference between them?

**Comment by akatsumata** [comment]
The difference is the build version.
It has a different number because it was recreated with fixes such as vulnerabilities.
Please check the date for the latest.

$ ls /usr/local/biotools/s/samtools:1.9* -l
-rw-r--r-- 1 root root 30167040 Sep  6  2019 /usr/local/biotools/s/samtools:1.9--h10a08f8_12
-rw-r--r-- 1 root root 26439680 Aug 25  2019 /usr/local/biotools/s/samtools:1.9--h43f6869_9
-rw-r--r-- 1 root root 23293952 Aug 25  2019 /usr/local/biotools/s/samtools:1.9--h46bd0b3_0
-rw-r--r-- 1 root root 26267648 Aug 31  2019 /usr/local/biotools/s/samtools:1.9--h57cc563_6
-rw-r--r-- 1 root root 26439680 Aug 25  2019 /usr/local/biotools/s/samtools:1.9--h57cc563_7
-rw-r--r-- 1 root root 27267072 Aug 31  2019 /usr/local/biotools/s/samtools:1.9--h8571acd_10
-rw-r--r-- 1 root root 27271168 Aug 25  2019 /usr/local/biotools/s/samtools:1.9--h8571acd_11
-rw-r--r-- 1 root root 25468928 Aug 25  2019 /usr/local/biotools/s/samtools:1.9--h8ee4bcc_1
-rw-r--r-- 1 root root 26267648 Aug 31  2019 /usr/local/biotools/s/samtools:1.9--h91753b0_2
-rw-r--r-- 1 root root 26267648 Aug 31  2019 /usr/local/biotools/s/samtools:1.9--h91753b0_3
-rw-r--r-- 1 root root 26267648 Aug 31  2019 /usr/local/biotools/s/samtools:1.9--h91753b0_4
-rw-r--r-- 1 root root 26267648 Aug 31  2019 /usr/local/biotools/s/samtools:1.9--h91753b0_5
-rw-r--r-- 1 root root 26439680 Aug 25  2019 /usr/local/biotools/s/samtools:1.9--h91753b0_8
$

**Comment by akatsumata** [comment, marked as answer]
The Singularity Container Image files provided by biocontainers are converted to Singularity Container Image files 
from Docker container image files on https://quay.io/organization/biocontainers. The Docker container image files 
on https://quay.io/organization/biocontainers is created based on that  https://github.com/BioContainers.

Looking at the contents of the Dockerfile, the analysis tool is often installed using conda install. Therefore, 
depending on when the Docker container image was built, the version of the software depending on the analysis tool may be different, 
or the Docker base image may be different. 
To guarantee reproducibility of the analysis, it is better to match the tag number of the Docker container image as well.

The meaning of the tag appears on the website, e.g. [quey.io](http://quey.io/) as follows.

https://quay.io/repository/biocontainers/samtools?tab=tags
